source/extensions/clusters/aggregate/cluster.cc

```diff
@@ -0,0 +1,150 @@
+#include "extensions/clusters/aggregate/cluster.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Clusters {
+namespace Aggregate {
+
+Cluster::Cluster(const envoy::api::v2::Cluster& cluster,
+                 const envoy::config::cluster::aggregate::ClusterConfig& config,
+                 Upstream::ClusterManager& cluster_manager, Runtime::Loader& runtime,
+                 Runtime::RandomGenerator& random,
+                 Server::Configuration::TransportSocketFactoryContext& factory_context,
+                 Stats::ScopePtr&& stats_scope, ThreadLocal::SlotAllocator& tls, bool added_via_api)
+    : Upstream::ClusterImplBase(cluster, runtime, factory_context, std::move(stats_scope),
+                                added_via_api),
+      cluster_manager_(cluster_manager), runtime_(runtime), random_(random),
+      tls_(tls.allocateSlot()) {
+  for (const auto& inner_cluster : config.clusters()) {
+    clusters_.emplace_back(inner_cluster);
+  }
+}
+
+PriorityContext
+Cluster::linearizePrioritySet(const std::function<bool(const std::string&)>& skip_predicate) {
+  Upstream::PrioritySetImpl priority_set;
+  std::vector<std::pair<uint32_t, Upstream::ThreadLocalCluster*>> priority_to_cluster;
```

Comment:
How come this isn't a map?

Reply:
The index in `priority_to_cluster` also matters. The index is the priority after linearization. The first element of the pair is the original priority before linearization.